workload/workloads/userProfile.go

```diff
@@ -71,8 +75,61 @@ func (w userProfile) Probabilities() [][]float64 {
 }
 
 func (w userProfile) Setup() error {
-	// TODO setup FTS index here for findRelatedProfile
 	gofakeit.Seed(int64(workload.RandSeed))
+
+	return createFtsIndex(w.scope)
+}
+
+func createFtsIndex(scope *gocb.Scope) error {
+	mgr := scope.SearchIndexes()
+	_, err := mgr.GetIndex("profile-statuses", nil)
+	if err == nil {
+		return nil
+	}
+
+	if !strings.Contains(err.Error(), "index not found") {
+		return errors.Wrap(err, "Failed to check if fts index has already been created")
+	}
+
+	index_definition := []byte(`{
+	 "name": "profile-statuses",
+	 "type": "fulltext-index",
+	 "params": {
+	  "doc_config": {
+	   "docid_prefix_delim": "",
+	   "docid_regexp": "",
+	   "mode": "scope.collection.type_field",
+	   "type_field": "type"
+	  },
+	  "mapping": {
+	   "default_mapping": {
+		"dynamic": true,
+		"enabled": false
+	   },
+	   "store_dynamic": true,
+	   "types": {
+		"identity.profiles": {
+		 "dynamic": true,
+		 "enabled": true
+		}
+	   }
+	  }
+	 },
+	 "sourceType": "gocbcore",
+	 "sourceName": "data"
+	}`)
+
+	var upsertIndex gocb.SearchIndex
+	err = json.Unmarshal(index_definition, &upsertIndex)
+	if err != nil {
+		return errors.Wrap(err, "Unmarshalling fts index definition failed")
```

Comment:
It looks like an error returned either here or from the next section will just have the test run anyway.  

I think this one should panic.  The next one might make sense to return and be handled there as a more graceful exit.

Reply:
This error will propagate up to workloads.go:Setup where it will panic.